platform-sdk/consensus-otter-tests/src/testFixtures/java/org/hiero/otter/fixtures/internal/logging/InMemoryAppender.java

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: Apache-2.0
+package org.hiero.otter.fixtures.internal.logging;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.apache.logging.log4j.core.Appender;
+import org.apache.logging.log4j.core.LogEvent;
+import org.apache.logging.log4j.core.appender.AbstractAppender;
+import org.apache.logging.log4j.core.config.Property;
+import org.apache.logging.log4j.core.config.plugins.Plugin;
+import org.apache.logging.log4j.core.config.plugins.PluginAttribute;
+import org.apache.logging.log4j.core.config.plugins.PluginFactory;
+import org.apache.logging.log4j.core.layout.PatternLayout;
+
+/**
+ * An {@link Appender} implementation for Log4j2 that provides in-memory storage
+ * for log events. This appender is used in testing to capture logs
+ * and validate them programmatically.
+ *
+ * @see AbstractAppender
+ */
+@Plugin(name = "InMemoryAppender", category = "Core", elementType = Appender.ELEMENT_TYPE)
+public class InMemoryAppender extends AbstractAppender {
+
+    private static final List<StructuredLog> logs = Collections.synchronizedList(new ArrayList<>());
+
+    /**
+     * Constructs an {@code InMemoryAppender} with the given name.
+     *
+     * @param name The name of the appender.
+     */
+    protected InMemoryAppender(String name) {
+        super(name, null, PatternLayout.createDefaultLayout(), false, Property.EMPTY_ARRAY);
```

Comment:
I am not familiar with log4j in detail. Maybe we can add a comment that explains the effect of these parameters.

Reply:
added comments